### ISSUE TITLE
Set FDT and cmdline args with extra_uEnv.txt

### ISSUE THREAD

**Issue by 20k-ultra** [balena-io/balena-supervisor#1385]
**Update**: Through various discussions it has been agreed that moving forward the Supervisor will no longer directly modify the devices `extlinux.conf` file. To simplify how kernal configuration values like FDT and additional cmdline args are managed the Supervisor will now be using `extra_uEnv.txt`. 

This allows us to get/set values and each device will intercept the values it supports from there.  For example, we just recently added support for setting FDT directive in extlinux.conf which required extending the [extlinux](https://github.com/balena-io/balena-supervisor/blob/master/src/config/backends/extlinux.ts) backend. In this new scenario we would simple take the configuration value from the target state provided by the cloud and set it in extra_uEnv.txt using the agreed key (in this case custom_fdt_file).

To achieve this we will just need to add a new config backend but we must keep in mind what will happen if a user is running an old balenaOS version but decides to upgrade the Supervisor. Since old versions do not support `extra_uEnv.txt` their target state won't be applied. 

---

As of balenaOS version v2.47.0+rev1 for Jetson devices the extlinux.conf has been moved to a read only partition and so the current strategy for modifying extlinux backend does not work for these versions. The change can be found in this [commit](https://github.com/balena-os/balena-jetson/commit/00c0a0a22b45552d998ca95af2628ced2955322b).

As a result, https://github.com/balena-os/balena-jetson/pull/79 has been created. It works by having the Supervisor append key values to `extra_uEnv.txt` in the boot partition. The currently purposed variable naming is:

```
custom_fdt_file=mycustom.dtb
extra_cmdline_args=ro rootwait isolcpus=3,4
```

Therefore, we must 
- (1) create a new strategy that will detect when Supervisor is running in this environment so that it can then 
- (2) append target state values (fdt, kernal cmds) in extra_uEnv.txt.

For environment detection it cannot rely on just checking the release version because balenaOS ESR does not use semver versioning.

**Comment by 20k-ultra** [comment]
Updated post to use new file and variable names.

**Comment by 20k-ultra** [comment]
Updated original post to contain new plan for solving this issue.

**Comment by acostach** [comment]
@20k-ultra we need to also note that the supervisor should NOT reboot the board due to any configuration change while HUP is running, or right after a reboot into the new OS, if the rollback files were not cleared yet.

**Comment by 20k-ultra** [comment]
@acostach  Can you create another issue for this please. Adding that will create more complexity to this issue and it is a separate issue.

**Comment by CameronDiver** [comment]
@acostach the supervisor shouldn't be running throughout the entire hup process. Adding more complexity to the supervisor doesn't make sense here as it doesn't control the hup, and it shouldnt know anything about it IMO. We need to make sure that the supervisor is stopped before starting and hup, and doesn't start until the hup is finished.
